src/nvme/types.h

```diff
@@ -7452,6 +7452,14 @@ enum nvme_admin_opcode {
 	nvme_admin_get_lba_status	= 0x86,
 };
 
+enum nvme_control_opcode {
+	nvme_control_pause		= 0x00,
+	nvme_control_resume		= 0x01,
+	nvme_control_abort		= 0x02,
+	nvme_control_get_state	= 0x03,
+	nvme_control_replay		= 0x04,
+};
```

Comment:
These declarations introduce some fairly generic names; I'd suggest making those mi-specific, similar to the MI-command opcode definitions: https://github.com/linux-nvme/libnvme/blob/55e29b85c9b85e69aff7b43dbd509b7be21b93ea/src/nvme/mi.h#L217-L222)

\- and I would suggest defining these in `mi.h` rather than the more-generic `types.h`

How about:

```c
enum nvme_mi_control_opcode {
	nvme_mi_control_opcode_pause		= 0x00,
	nvme_mi_control_opcode_resume		= 0x01,
	nvme_mi_control_opcode_abort		= 0x02,
	nvme_mi_control_opcode_get_state	= 0x03,
	nvme_mi_control_opcode_replay		= 0x04,
}
```

Reply:
nice, done